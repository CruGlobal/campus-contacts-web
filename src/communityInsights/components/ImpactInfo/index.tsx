import { useTranslation } from 'react-i18next';
import { useQuery } from 'react-apollo-hooks';
import React from 'react';

import Header from '../Header';

interface Props {
    query: any;
    text: (data: any) => string;
    variables: any;
}

const ImpactInfo = ({ query, text, variables }: Props) => {
    const { data, loading } = useQuery(query, { variables });
    const { t } = useTranslation('insights');

    if (loading) {
        return <Header>{t('gatheringStats')}</Header>;
    }
    return <Header>{text(data)} </Header>;
};

export default ImpactInfo;
