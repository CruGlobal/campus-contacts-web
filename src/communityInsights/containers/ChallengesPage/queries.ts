import gql from 'graphql-tag';

const GET_IMPACT_CHALLENGES = gql`
    query impactReportStepsCount($id: ID!) {
        community(id: $id) {
            impactReport {
                stepsCount
            }
        }
    }
`;

const GET_CHALLENGES = gql`
    query communityChallenges(
        $id: ID!
        $first: Int
        $last: Int
        $after: String
        $before: String
        $sortBy: [CommunityChallengesSortEnum!]
    ) {
        community(id: $id) {
            communityChallenges(
                first: $first
                last: $last
                after: $after
                before: $before
                sortBy: $sortBy
            ) {
                nodes {
                    id
                    title
                    acceptedCount
                    completedCount
                    createdAt
                    endDate
                }
                pageInfo {
                    hasNextPage
                    startCursor
                    endCursor
                    hasPreviousPage
                }
            }
        }
    }
`;

export { GET_CHALLENGES, GET_IMPACT_CHALLENGES };
