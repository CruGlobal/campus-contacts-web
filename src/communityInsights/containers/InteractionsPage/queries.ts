import gql from 'graphql-tag';

const INTERACTIONS_TOTAL_REPORT = gql`
    query impactReportInteractionsCount($id: ID!) {
        community(id: $id) {
            impactReport {
                interactionsCount
                interactionsReceiversCount
            }
        }
    }
`;

const INTERACTIONS_TOTAL_COMPLETED_REPORT = gql`
    query communityReportInteractions(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                interactions {
                    nodes {
                        interactionCount
                        interactionType {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const INTERACTIONS_COMPLETED_REPORT = gql`
    query communityReportDaysInteractions(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                daysReport {
                    nodes {
                        date
                        interactionsCount
                        interactionResults {
                            nodes {
                                count
                                interactionType {
                                    name
                                }
                            }
                        }
                    }
                }
            }
        }
    }
`;

export {
    INTERACTIONS_TOTAL_REPORT,
    INTERACTIONS_TOTAL_COMPLETED_REPORT,
    INTERACTIONS_COMPLETED_REPORT,
};
