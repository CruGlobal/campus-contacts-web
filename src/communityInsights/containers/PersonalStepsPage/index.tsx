import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import moment from 'moment';

import Card from '../../components/Card';
import { AppContext } from '../../../appContext';
import StagesSummary from '../../components/StagesSummary';
import StepsChart from '../../components/StepsChart';
import ImpactInfo from '../../components/ImpactInfo';
import FiltersChart from '../../components/FiltersChart';

import {
    GET_IMPACT_REPORT_MOVED,
    GET_IMPACT_REPORT_STEPS_TAKEN,
    GET_STAGES_REPORT_MEMBER_COUNT,
    GET_STEPS_COMPLETED_REPORT,
    GET_STAGES_REPORT_STEPS_ADDED,
    GET_TOTAL_STEPS_COMPLETED_SUMMARY,
} from './queries';
// GET_IMPACT_REPORT_STEPS_TAKEN INTERFACES
import { impactReportPersonalStepsCompletedCount } from './__generated__/impactReportPersonalStepsCompletedCount';
// GET_TOTAL_STEPS_COMPLETED_SUMMARY INTERFACES
import {
    communityReportStagesPersonalStepsCompleted,
    communityReportStagesPersonalStepsCompleted_community_report_stagesReport as communityReportStagesPersonalStepsCompletedEntry,
} from './__generated__/communityReportStagesPersonalStepsCompleted';
// GET_STEPS_COMPLETED_REPORT INTERFACES
import {
    communityReportDaysPersonalSteps,
    communityReportDaysPersonalSteps_community_report_daysReport as communityReportDaysPersonalRow,
    communityReportDaysPersonalSteps_community_report_daysReport_stageResults as communityReportDaysPersonalStage,
} from './__generated__/communityReportDaysPersonalSteps';
// GET_STAGES_REPORT_STEPS_ADDED INTERFACES
import {
    communityReportStagesPersonalStepsAdded,
    communityReportStagesPersonalStepsAdded_community_report_stagesReport as communityReportStagesPersonalStepsAddedRow,
} from './__generated__/communityReportStagesPersonalStepsAdded';
// GET_IMPACT_REPORT_MOVED INTERFACES
import { impactReportStageProgressionCount } from './__generated__/impactReportStageProgressionCount';
// GET_STAGES_REPORT_MEMBER_COUNT INTERFACES
import {
    communityReportStagesPersonalMemberCount,
    communityReportStagesPersonalMemberCount_community_report_stagesReport as communityReportStagesPersonalMemberCountRow,
} from './__generated__/communityReportStagesPersonalMemberCount';

const PersonalStepsPage = () => {
    const { t } = useTranslation('insights');
    const { orgId } = useContext(AppContext);

    return (
        <div>
            <ImpactInfo
                query={GET_IMPACT_REPORT_STEPS_TAKEN}
                text={(report: impactReportPersonalStepsCompletedCount) =>
                    t('personalSteps.taken', {
                        count:
                            report.community.impactReport
                                .personalStepsCompletedCount,
                    })
                }
                variables={{ id: orgId }}
            />
            <Card title={t('personalSteps.completedTotal')}>
                <StagesSummary
                    query={GET_TOTAL_STEPS_COMPLETED_SUMMARY}
                    variables={{ id: orgId }}
                    mapData={(
                        data: communityReportStagesPersonalStepsCompleted,
                    ) =>
                        data.community.report.stagesReport.map(
                            (
                                entry: communityReportStagesPersonalStepsCompletedEntry,
                            ) => ({
                                stage: entry.stage.name,
                                icon: entry.stage.name
                                    .toLowerCase()
                                    .replace(' ', '-'),
                                count: entry.personalStepsCompletedCount,
                            }),
                        )
                    }
                />
            </Card>
            <Card
                title={t('personalSteps.completed')}
                subtitle={t('personalSteps.completedSubtitle')}
            >
                <FiltersChart
                    query={GET_STEPS_COMPLETED_REPORT}
                    variables={{ id: orgId }}
                    mapData={(data: communityReportDaysPersonalSteps) =>
                        data.community.report.daysReport.map(
                            (row: communityReportDaysPersonalRow) => ({
                                ['total']: row.personalStepsCompletedCount,
                                ['stages']: row.stageResults.map(
                                    (
                                        stage: communityReportDaysPersonalStage,
                                    ) => ({
                                        name: stage.stage.name,
                                        count:
                                            stage.personalStepsCompletedCount,
                                    }),
                                ),
                                ['date']: row.date,
                            }),
                        )
                    }
                    currentDate={moment().toDate()}
                    label={t('personalSteps.legend')}
                />
            </Card>
            <Card
                title={t('personalSteps.added')}
                subtitle={t('personalSteps.addedSubtitle')}
            >
                <StepsChart
                    query={GET_STAGES_REPORT_STEPS_ADDED}
                    mapData={(data: communityReportStagesPersonalStepsAdded) =>
                        data.community.report.stagesReport.map(
                            (
                                row: communityReportStagesPersonalStepsAddedRow,
                            ) => ({
                                [t(
                                    'personalSteps.label',
                                )]: row.personalStepsAddedCount,
                                [t('stage')]: row.stage.name.toUpperCase(),
                            }),
                        )
                    }
                    label={t('personalSteps.label')}
                    index={t('stage')}
                    variables={{
                        period: 'P10Y',
                        id: orgId,
                        endDate: moment().format(),
                    }}
                />
            </Card>

            <ImpactInfo
                query={GET_IMPACT_REPORT_MOVED}
                text={(report: impactReportStageProgressionCount) =>
                    t('personalSteps.reached', {
                        count:
                            report.community.impactReport.stageProgressionCount,
                    })
                }
                variables={{ id: orgId }}
            />
            <Card
                title={t('personalSteps.members')}
                subtitle={t('personalSteps.membersSubtitle')}
            >
                <StepsChart
                    query={GET_STAGES_REPORT_MEMBER_COUNT}
                    mapData={(data: communityReportStagesPersonalMemberCount) =>
                        data.community.report.stagesReport.map(
                            (
                                row: communityReportStagesPersonalMemberCountRow,
                            ) => ({
                                [t('members')]: row.memberCount,
                                [t('stage')]: row.stage.name.toUpperCase(),
                            }),
                        )
                    }
                    label={t('members')}
                    index={t('stage')}
                    variables={{
                        period: 'P10Y',
                        id: orgId,
                        endDate: moment().format(),
                    }}
                />
            </Card>
        </div>
    );
};

export default PersonalStepsPage;
