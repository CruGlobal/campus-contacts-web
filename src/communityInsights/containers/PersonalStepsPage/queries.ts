import gql from 'graphql-tag';

const GET_IMPACT_REPORT_MOVED = gql`
    query impactReportStageProgressionCount($id: ID!) {
        community(id: $id) {
            impactReport {
                stageProgressionCount
            }
        }
    }
`;

const GET_IMPACT_REPORT_STEPS_TAKEN = gql`
    query impactReportPersonalStepsCompletedCount($id: ID!) {
        community(id: $id) {
            impactReport {
                personalStepsCompletedCount
            }
        }
    }
`;

const GET_STAGES_REPORT_MEMBER_COUNT = gql`
    query communityReportStagesPersonalMemberCount(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        memberCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const GET_STAGES_REPORT_STEPS_ADDED = gql`
    query communityReportStagesPersonalStepsAdded(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        personalStepsAddedCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const GET_STEPS_COMPLETED_REPORT = gql`
    query communityReportDaysPersonalSteps(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                daysReport {
                    nodes {
                        date
                        personalStepsCompletedCount
                        stageResults {
                            nodes {
                                personalStepsCompletedCount
                                stage {
                                    name
                                }
                            }
                        }
                    }
                }
            }
        }
    }
`;

const GET_TOTAL_STEPS_COMPLETED_SUMMARY = gql`
    query communityReportStagesPersonalStepsCompleted(
        $id: ID!
        $period: String!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        personalStepsCompletedCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

export {
    GET_IMPACT_REPORT_MOVED,
    GET_IMPACT_REPORT_STEPS_TAKEN,
    GET_STAGES_REPORT_MEMBER_COUNT,
    GET_STEPS_COMPLETED_REPORT,
    GET_STAGES_REPORT_STEPS_ADDED,
    GET_TOTAL_STEPS_COMPLETED_SUMMARY,
};
