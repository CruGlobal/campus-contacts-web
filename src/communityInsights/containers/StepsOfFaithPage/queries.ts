import gql from 'graphql-tag';

const GET_IMPACT_REPORT_TAKEN = gql`
    query impactReportOtherStepsCompleted($id: ID!) {
        community(id: $id) {
            impactReport {
                othersStepsCompletedCount
                othersStepsReceiversCompletedCount
            }
        }
    }
`;

const GET_TOTAL_STEPS_COMPLETED_REPORT = gql`
    query communityReportStagesOthersStepsCompleted(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        othersStepsCompletedCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const GET_STEPS_COMPLETED_REPORT = gql`
    query communityReportDaysOthersSteps(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                daysReport {
                    nodes {
                        date
                        othersStepsCompletedCount
                        stageResults {
                            nodes {
                                othersStepsCompletedCount
                                stage {
                                    name
                                }
                            }
                        }
                    }
                }
            }
        }
    }
`;

const GET_STAGES_REPORT = gql`
    query communityReportStagesOthersStepsAdded(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        othersStepsAddedCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const GET_IMPACT_REPORT_REACHED = gql`
    query impactReportMembersStageProgressionCount($id: ID!) {
        community(id: $id) {
            impactReport {
                membersStageProgressionCount
            }
        }
    }
`;

const GET_STAGES_PEOPLE_REPORT = gql`
    query communityReportStagesOthersContactCount(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        contactCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

export {
    GET_IMPACT_REPORT_TAKEN,
    GET_IMPACT_REPORT_REACHED,
    GET_TOTAL_STEPS_COMPLETED_REPORT,
    GET_STEPS_COMPLETED_REPORT,
    GET_STAGES_REPORT,
    GET_STAGES_PEOPLE_REPORT,
};
