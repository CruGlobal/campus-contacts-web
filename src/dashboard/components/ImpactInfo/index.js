import { useTranslation } from 'react-i18next';
import { useQuery } from 'react-apollo-hooks';
import React from 'react';

import Header from '../Header';

const ImpactInfo = ({ query, text, variables }) => {
    const { data, loading } = useQuery(query, { variables });
    const { t } = useTranslation('insights');

    if (loading) {
        return <Header>{t('loading')}</Header>;
    }
    return <Header>{text(data)} </Header>;
};

export default ImpactInfo;
