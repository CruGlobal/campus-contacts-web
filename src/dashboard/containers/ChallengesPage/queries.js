import gql from 'graphql-tag';

const GET_IMPACT_CHALLENGES = gql`
    query communityReport {
        communityReport {
            impactReport {
                interactionsCount
                interactionsReceiversCount
                challengesCount
            }
        }
    }
`;

const GET_CHALLENGES = gql`
    query globalCommunityChallenges($id: ID!, $first: Int, $after: String) {
        community(id: $id) {
            communityChallenges(first: $first, after: $after) {
                nodes {
                    id
                    title
                    acceptedCount
                    completedCount
                    createdAt
                    endDate
                }
                pageInfo {
                    hasNextPage
                    startCursor
                    endCursor
                    hasPreviousPage
                }
            }
        }
    }
`;

export { GET_CHALLENGES, GET_IMPACT_CHALLENGES };
