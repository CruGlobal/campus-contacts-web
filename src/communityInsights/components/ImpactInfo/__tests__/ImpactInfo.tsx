import React from 'react';
import gql from 'graphql-tag';
import { waitForElement } from '@testing-library/react';

import { renderWithContext } from '../../../../testUtils';
import ImpactInfo from '..';
import { GET_IMPACT_REPORT_MOVED } from '../../../containers/PersonalStepsPage/queries';

describe('<ImpactInfo />', () => {
    it('should render properly loading state', () => {
        const { snapshot, unmount } = renderWithContext(
            <ImpactInfo
                query={GET_IMPACT_REPORT_MOVED}
                text={report =>
                    `Text with value ${report.community.impactReport.stageProgressionCount}`
                }
                variables={{ id: 1 }}
            />,
            {
                appContext: {
                    orgId: 1,
                },
            },
        );
        snapshot();
        unmount();
    });

    it('should render properly with data', async () => {
        const { snapshot, getByText } = renderWithContext(
            <ImpactInfo
                query={GET_IMPACT_REPORT_MOVED}
                text={report =>
                    `Text with value ${report.community.impactReport.stageProgressionCount}`
                }
                variables={{ id: 1 }}
            />,
            {
                mocks: {
                    Query: () => ({
                        community: () => ({
                            impactReport: () => ({
                                stageProgressionCount: () => 15,
                            }),
                        }),
                    }),
                },
                appContext: {
                    orgId: 1,
                },
            },
        );

        await waitForElement(() =>
            getByText('Text with value', { exact: false }),
        );
        snapshot();
    });
});
