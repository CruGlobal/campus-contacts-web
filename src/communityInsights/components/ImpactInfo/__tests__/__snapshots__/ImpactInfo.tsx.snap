// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImpactInfo /> should render properly loading state 1`] = `
.emotion-0 {
  font-weight: 300;
  font-size: 30px;
  line-height: 38px;
  color: #505256;
  margin-bottom: 0px;
  margin-top: 48px;
  padding-left: 21px;
}

<div>
  <h1
    class="emotion-0"
  >
    Gathering your stats...
  </h1>
</div>
`;

exports[`<ImpactInfo /> should render properly with data 1`] = `
.emotion-0 {
  font-weight: 300;
  font-size: 30px;
  line-height: 38px;
  color: #505256;
  margin-bottom: 0px;
  margin-top: 48px;
  padding-left: 21px;
}

<div>
  <h1
    class="emotion-0"
  >
    Text with value 15
     
  </h1>
</div>
`;
